docs/source/notes/extending.rst

```diff
@@ -98,6 +98,15 @@ non-Variable arguments::
             # Gradients of non-Tensor arguments to forward must be None.
             return grad_output * ctx.constant, None
 
+.. note::
+    To implement custom double backward (i.e., second order derivatives) for a
+    custom :class:`~torch.autograd.function`, e.g., named ``MyFunction``, we can
+    define another custom  :class:`~torch.autograd.function`, e.g., named
+    ``MyFunctionBackward``, where the ``MyFunctionBackward.forward`` method
+    implements the backward pass of ``MyFunction`` and is called in
+    ``MyFunction.backward`` method, and ``MyFunctionBackward.backward`` method
+    implements the custom double backward computation for ``MyFunction``.
```

Comment:
This seems like a needlessly complicated description. Basically, just point out that the inputs to `backward` are honest to goodness variables, so if you implement `backward` using differentiable operations (e.g., `MyFunctionBackward` invocation), higher order derivatives will work.